presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ArrayColumnValidator.java

```diff
@@ -67,22 +68,41 @@ public ArrayColumnValidator()
             checksum = new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(column.getIdentifier()));
         }
 
-        return ImmutableList.of(new SingleColumn(checksum, Optional.of(delimitedIdentifier(getChecksumColumnAlias(column)))));
+        Expression arrayCardinalitySum = new CoalesceExpression(
+                new FunctionCall(
+                        QualifiedName.of("sum"),
+                        ImmutableList.of(new FunctionCall(QualifiedName.of("cardinality"), ImmutableList.of(column.getIdentifier())))),
+                new LongLiteral("0"));
+
+        return ImmutableList.of(
+                new SingleColumn(checksum, Optional.of(delimitedIdentifier(getChecksumColumnAlias(column)))),
+                new SingleColumn(arrayCardinalitySum, Optional.of(delimitedIdentifier(getCardinalitySumColumnAlias(column)))));
     }
 
     @Override
     public ColumnMatchResult validate(Column column, ChecksumResult controlResult, ChecksumResult testResult)
     {
         String sortedChecksumColumnAlias = getChecksumColumnAlias(column);
-        Object controlSortedChecksum = controlResult.getChecksum(sortedChecksumColumnAlias);
-        Object testSortedChecksum = testResult.getChecksum(sortedChecksumColumnAlias);
+        Object controlChecksum = controlResult.getChecksum(sortedChecksumColumnAlias);
+        Object testChecksum = testResult.getChecksum(sortedChecksumColumnAlias);
+
+        String cardinalitySumColumnAlias = getCardinalitySumColumnAlias(column);
+        Object controlCardinalitySum = controlResult.getChecksum(cardinalitySumColumnAlias);
+        Object testCardinalitySum = testResult.getChecksum(cardinalitySumColumnAlias);
+
         return new ColumnMatchResult(
-                Objects.equals(controlSortedChecksum, testSortedChecksum),
-                format("control(checksum: %s) test(checksum: %s)", controlSortedChecksum, testSortedChecksum));
+                Objects.equals(controlChecksum, testChecksum) && Objects.equals(controlCardinalitySum, testCardinalitySum),
+                format("control(checksum: %s, cardinality_sum: %s) test(checksum: %s, cardinality_sum: %s)",
+                        controlChecksum, controlCardinalitySum, testChecksum, testCardinalitySum));
```

Comment:
Merge line 95 and 96, or, one line per parameter:
```Java
format(
    "control(checksum: %s, cardinality_sum: %s) test(checksum: %s, cardinality_sum: %s)",
    controlChecksum,
    controlCardinalitySum,
    testChecksum,
    testCardinalitySum));
```

Reply:
Airlift auto reformat results in a different indent as shown in the latest diff.